cmake/templates/python_win32_wrapper.cpp

```diff
@@ -0,0 +1,177 @@
+//
+// Software License Agreement (BSD License)
+//
+// Copyright (c) 2011, Yujin Robot
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//
+// * Redistributions of source code must retain the above copyright
+//   notice, this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above
+//   copyright notice, this list of conditions and the following
+//   disclaimer in the documentation and/or other materials provided
+//   with the distribution.
+// * Neither the name of Yujin Robot nor the names of its
+//   contributors may be used to endorse or promote products derived
+//   from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+
+/*
+ * @file python_win32_wrapper.cpp
+ *
+ * @brief Launch a Python script located in the same location and have the same name as the compiled executable
+ *
+ * @date March 2011
+ */
+
+#ifdef _WIN32
+
+#include <iostream>
+#include <windows.h>
+
+int main(int argc, char* argv[]) try
+{
+    const auto GetCurrentModuleName = []() -> std::string
+    {
+        // initialize buffer string with at least one character
+        std::string moduleName = " ";
+        while (true)
+        {
+            // retrieves the path of the executable file of the current process
+            auto result = ::GetModuleFileName(nullptr, &moduleName[0], static_cast<DWORD>(moduleName.size()));
+            if (!result)
+            {
+                throw ::GetLastError();
+            }
+            else if (result == moduleName.size() && ::GetLastError() == ERROR_INSUFFICIENT_BUFFER)
+            {
+                // buffer not large enough
+                moduleName.resize(moduleName.size() * 2);
+                continue;
```

Comment:
Is this incremental increasing of the size really necessary? Why not allocate a buffer of 1k and call it a day (and if the result doesn't fit in error out)?

Reply:
> Is this incremental increasing of the size really necessary?

it's not critical, but I think it's nice to have since this could make use of the `std::string` container, and doesn't matter how long the buffer is.

`MAX_PATH` is only 260, and is becoming meaningless in the current day. since the size of the buffer doubles every time, it would only take 10 cycles to reach >1k, shouldn't hurt performance either =)

Reply:
While it might not hurt performance it does hurt readability and maintainability of the code. So allocating `MAX_PATH` sounds like the right choice.

Reply:
> While it might not hurt performance it does hurt readability and maintainability of the code. So allocating `MAX_PATH` sounds like the right choice.

change to use `MAX_PATH` instead